huaweicloud/services/nat/data_source_huaweicloud_nat_private_gateways.go

```diff
@@ -213,16 +220,23 @@ func flattenListGatewaysResponseBodyGateways(resp interface{}) []interface{} {
 			"vpc_id":                utils.PathSearch("downlink_vpcs[0].vpc_id", v, nil),
 			"subnet_id":             utils.PathSearch("downlink_vpcs[0].virsubnet_id", v, nil),
 			"enterprise_project_id": utils.PathSearch("enterprise_project_id", v, nil),
+			"tags":                  utils.FlattenTagsToMap(utils.PathSearch("tags", v, nil)),
 		})
 	}
 	return rst
 }
 
 func filterListGatewaysResponseBodyGateways(all []interface{}, d *schema.ResourceData) []interface{} {
```

Comment:
```suggestion
func filterListGatewaysResponseByTags(all []interface{}, d *schema.ResourceData) []interface{} {
```
This function is used only to filter results by tags. The current name may cause ambiguity.

Reply:
done